vault/external_tests/activity/activity_test.go

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package activity
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/vault/helper/namespace"
+	"github.com/hashicorp/vault/helper/timeutil"
+	vaulthttp "github.com/hashicorp/vault/http"
+	"github.com/hashicorp/vault/vault"
+	"github.com/hashicorp/vault/vault/activity"
+)
+
+// TestActivityLog_DefaultBillingPeriodQuery verifies that
+// when the correct startTime and endTime values are used
+// during requests to /sys/internal/counters/activity.
+// This tests checks scenarios where various optional values
+// remain unset, a verifies that the correct default date ranges are
+// applied. This also serves as a regression test for the deprecated
+// "default_report_months" field.
+func TestActivityLog_DefaultBillingPeriodQuery(t *testing.T) {
+	timeutil.SkipAtEndOfMonth(t)
+
+	ctx := context.Background()
+
+	// Create license where start date is set to two years ago
+	thisMonth := timeutil.StartOfMonth(time.Now().UTC())
+
+	// Setup a test cluster with root only
+	coreConfig := &vault.CoreConfig{
+		ActivityLogConfig: vault.ActivityLogCoreConfig{
+			DisableTimers: true,
+		},
+	}
+	cluster := vault.NewTestCluster(t, coreConfig, &vault.TestClusterOptions{
+		HandlerFunc: vaulthttp.Handler,
+		NumCores:    1,
+	})
+	cluster.Start()
+	defer cluster.Cleanup()
+
+	client := cluster.Cores[0].Client
+	core := cluster.Cores[0].Core
+
+	// Set up 48 months of data, all ending last month
+	lastMonth := thisMonth.AddDate(0, -1, 0)
+
+	testData := make([]*activity.PrecomputedQuery, 48)
+	for m := range testData {
+		testData[m] = &activity.PrecomputedQuery{
+			StartTime: timeutil.StartOfMonth(thisMonth.AddDate(0, -48+m, 0)),
+			EndTime:   timeutil.EndOfMonth(thisMonth.AddDate(0, -1, 0)),
+			Namespaces: []*activity.NamespaceRecord{
+				{
+					NamespaceID:     namespace.RootNamespaceID,
+					Entities:        uint64(5 + m),
+					NonEntityTokens: uint64(10 + m),
+				},
+			},
+		}
+		t.Logf("Injecting %v -- %v", testData[m].StartTime, testData[m].EndTime)
+		err := core.ActivityLogInjectResponse(ctx, testData[m])
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
```

Comment:
I'm not 100% certain, but I don't think this should be needed. The activity export should display empty months, but will also include the field that we're after (default_report_months).

Reply:
oops this was mistake, I removed this change (should be in another pr instead)